src/main/java/org/opensrp/service/EventService.java

```diff
@@ -424,9 +424,71 @@ private void triggerPlanEvaluation(Event event, String username) {
 		String planIdentifier = event.getDetails() != null ? event.getDetails().get("planIdentifier") : null;
 		if (isPlanEvaluationEnabled && planIdentifier != null) {
 			PlanDefinition plan = planRepository.get(planIdentifier);
-			if (plan.getStatus().equals(PlanDefinition.PlanStatus.ACTIVE) && (plan.getEffectivePeriod().getEnd() == null
+			if (plan != null && plan.getStatus().equals(PlanDefinition.PlanStatus.ACTIVE) && (plan.getEffectivePeriod().getEnd() == null
 					|| plan.getEffectivePeriod().getEnd().isAfter(LocalDate.now().toDateTimeAtStartOfDay())))
 				taskGenerator.processPlanEvaluation(plan, username, event);
 		}
 	}
+
+	public ExportEventDataSummary exportEventData(String planIdentifier, String eventType, Date fromDate, Date toDate)
+			throws JsonProcessingException {
+		List<org.opensrp.domain.postgres.Event> pgEvents = allEvents.getEventData(planIdentifier, eventType, fromDate, toDate);
+		ExportEventDataSummary exportEventDataSummary = new ExportEventDataSummary();
+		List<List<Object>> allRows = new ArrayList<>();
+		boolean returnHeader = true;
+		Map<String, String> columnNamesAndLabels = exportEventDataMapper.getColumnNamesAndLabelsByEventType(eventType);
+		boolean settingsExist = columnNamesAndLabels != null && columnNamesAndLabels.size() > 0 ? true : false;
+
+		if (settingsExist)
+			allRows.add(exportEventDataMapper
+					.getExportEventDataAfterMapping(null, eventType, returnHeader, settingsExist)); //for header row
+
+			//Assumption : All pgEvents would have similar obs fields to include as a header
+		else
+			allRows.add(exportEventDataMapper.getExportEventDataAfterMapping(
+					pgEvents != null && pgEvents.size() > 0 && pgEvents.get(0) != null ? pgEvents.get(0).getJson() : "",
```

Comment:
There is some redundant checks here, pgEvents cannot be null if being returned by mapper. 

Also `pgEvents.size()` > will mean this can never happen `pgEvents.get(0) != null`

Reply:
Removed these redundant checks